### PR TITLE
Align peerdeps

### PR DESCRIPTION
**Pull request by sndrs**
## What is the purpose of this change?

bring the way we use peerdeps up to speed

## What does this change?

- eslint plugins express eslint version they need
- packages express the widest possible range of peerdeps
- workspace installs the lowest possible version of dev deps that our packages also have peerdeps on

### CODE REVIEW

File: packages/@guardian/source-react-components/package.json [hunk 1]
```diff
@@ -24,9 +24,9 @@
     "rollup-plugin-ts": "^1.4.7"
   },
   "peerDependencies": {
-    "@emotion/react": "^11.1.2",
-    "@guardian/source-foundations": "^4.0.0-rc.1",
-    "react": "^17.0.1"
+    "@emotion/react": "^11.0.0",
+    "@guardian/source-foundations": "4.0.0-rc.5",
+    "react": "^17.0.0"
```
Comment (SiAdcock) [inline packages/@guardian/source-react-components/package.json:29]: This is cool and specific and I love it 😊 For posterity, is there a reason you couldn't go with `17.x` as the semver range?

**Review by SiAdcock — APPROVED** [review]
Yeeees 💯

File: packages/@guardian/source-react-components/package.json (hunk 1 above)
Comment (sndrs) [reply packages/@guardian/source-react-components/package.json:29]: Just for consistency with any minor/patch version minimums, as with prettier in configs. Suddenly you need to go from 4.x to ^4.1.0. Going with the one format seemed easier. Do you prefer the x?

Comment (SiAdcock) [reply packages/@guardian/source-react-components/package.json:29]: No, I much prefer the consistency here. Good call 👍

**Pull request merged**